onnx/defs/training/defs.cc

```diff
@@ -0,0 +1,171 @@
+// Copyright (c) ONNX Project Contributors.
+// Licensed under the MIT license.
+
+#include <algorithm>
+#include <cmath>
+#include "onnx/defs/function.h"
+#include "onnx/defs/schema.h"
+
+namespace ONNX_NAMESPACE {
+static const char* Gradient_ver12_doc = R"DOC(
+Gradient operator computes the derivatives of some tensors with respect to
+a specified tensor. This operator is widely used in gradient-based training
+algorithms. For a concrete example, let's consider a computation graph,
+
+```
+X -----.
+       |
+       v
+W --> Conv --> H --> Gemm --> Y
+                      ^
+                      |
+                      Z
+```
+
+, where W and Z are trainable tensors. Note that operators' attributes are
+omitted for the sake of simplicity. Let dY/dW (dY/dZ) be the gradient of
+Y with respect to W (Z). The user can compute gradient by inserting Gradient
+operator to form
+
+```
+W --> Conv --> H --> Gemm --> Y
+|      ^              ^
+|      |              |
+|      X              Z
+|      |              |
+|      |   .----------'
+|      |   |  (X/W/Z is the 1st/2nd/3rd input of Gradient as shown in "xs")
+|      v   v
+'---> Gradient(xs=["X", "W", "Z"], y=["Y"]) ---> dY/dX (1st output of Gradient)
```

Comment:
Why is "y" shown as a list here? It is a single string according to the definition below.

Reply:
Fixed. Thanks a lot for poiniting this out.